src/AbstractUri.php

```diff
@@ -362,33 +356,76 @@ protected static function urlEncodeMatch(array $matches): string
      *
      * @return string|null
      */
-    protected function formatHost($host)
+    protected function formatHost(string $host = null)
     {
-        static $pattern = '/^(?<name>[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)(?:\.(?&name))*$/i';
-
-        if ('' == $host || false !== strpos($host, ']')) {
+        if (null === $host || '' === $host) {
             return $host;
         }
 
+        static $pattern = '/^(?<name>[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)(?:\.(?&name))*$/i';
```

Comment:
As we learned, hosts don't need to comply to DNS rules, therefore the limit of label lengths is unnecessary. This should just be a regex that checks for `reg-name` I think.

Reply:
indeed but adding this check improve the performance for the majority of cases. Most host do use the DNS system. So it's a optimization for the common usage.

Reply:
Doesn't the below `is_host` catch those?